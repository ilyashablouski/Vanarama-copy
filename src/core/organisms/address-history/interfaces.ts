import { IBaseProps } from '../../interfaces/base';
import { IDropdown } from '../../interfaces/dropdown';
import { IAddress } from '../../interfaces/address';

interface IPropertyStatuses extends IDropdown {}

export interface IAddressHistoryObject {
  address: IAddress;
  propertyStatus: string;
  month: number;
  year: number;
}

export interface IAddressHistoryProps extends IBaseProps {
  name: string;
  minYearsOfHistory: number;
  lookBackYears: number;
  propertyStatuses: IPropertyStatuses;
  addressHistory?: IAddressHistoryObject[];
  setValue?: Function; // Callback.
}

export interface IAddressHistoryState {
  addressHistory?: IAddressHistoryObject[];
  completed: boolean;
  allValid: boolean;
}
