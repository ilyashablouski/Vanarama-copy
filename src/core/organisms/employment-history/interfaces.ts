import { IBaseProps } from '../../interfaces/base';
import { IDropdown } from '../../interfaces/dropdown';
import { IAddress } from '../../interfaces/address';

interface IEmploymentStatuses extends IDropdown {}

export interface IEmploymentHistoryObjectSlim {
  employmentStatus: string;
  month: number | null;
  year: number | null;
}

export interface IEmploymentHistoryObjectFull
  extends IEmploymentHistoryObjectSlim {
  contractType: string;
  jobTitle: string;
  companyName: string;
  phoneNumber: string;
  address: IAddress | null;
  income: string;
}

export type TEmploymentHistory =
  | IEmploymentHistoryObjectSlim
  | IEmploymentHistoryObjectFull;

export interface IEmploymentHistoryProps extends IBaseProps {
  minYearsOfHistory: number;
  lookBackYears: number;
  employmentStatuses: IEmploymentStatuses;
  employmentHistory?: TEmploymentHistory[];
  setValue?: Function; // Callback.
}

export interface IEmploymentHistoryState {
  employmentHistory: TEmploymentHistory[];
  completed: boolean;
  allValid: boolean;
}
