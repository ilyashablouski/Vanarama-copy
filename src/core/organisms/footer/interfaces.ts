import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IFooterProps extends IBaseProps {
  emailAddress?: string;
  phoneNumber?: string;
  openingTimes?: IOpeningTimes[];
  headingZero?: string;
  headingOne?: string;
  headingTwo?: string;
  headingThree?: string;
  linkGroupOne?: ILink[];
  linkGroupTwo?: ILink[];
  linkGroupThree?: ILink[];
  legalText?: string;
}

export interface IOpeningTimes {
  day: string;
  time: string;
}
