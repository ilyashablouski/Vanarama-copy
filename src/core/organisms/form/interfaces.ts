import { FormHTMLAttributes } from 'react';
import { IBaseProps } from '../../interfaces/base';

export interface IFormProps
  extends IBaseProps,
    FormHTMLAttributes<HTMLFormElement> {
  children: React.ReactNode;
  /**
   * Whether there a validation errors in this form
   */
  invalid?: boolean;
  /**
   * The function to call when the form is submitted
   */
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}

export interface IFormErrorProps extends IBaseProps {
  children: React.ReactNode;
}
