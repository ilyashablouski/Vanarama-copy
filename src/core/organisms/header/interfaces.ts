import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IHeaderProps extends IBaseProps {
  topBarLinks?: ILink[];
  loginLink?: ILink;
  phoneNumber?: string;
  showIvan?: boolean;
  message?: string;
}
