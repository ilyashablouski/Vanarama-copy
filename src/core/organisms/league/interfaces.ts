import { IBaseProps } from '../../interfaces/base';

export interface ILeagueProps extends IBaseProps {
  link: string;
  altText?: string;
}
