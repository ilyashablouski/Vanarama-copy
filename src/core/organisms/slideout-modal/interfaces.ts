import { MouseEvent } from 'react';

import { IBaseProps } from '../../interfaces/base';

export interface INavItemProps {
  label: string;
  selected?: string[];
  onClick?(e: MouseEvent): void;
}

export interface IPanelMainMenu {
  setOpenPanel(e: string | null): void;
  searchFilters: ISearchFilters;
  selectedFilters?: ISelectedFilters;
}

export interface IPanelHeaderProps extends IFilter {
  panelId: string;
  label: string;
  setOpenPanel(e: string | null): void;
}

export interface IPanelBody extends IFilter {
  panelId: string;
}

export interface IPanelProps extends IFilter {
  panelId: string;
  setOpenPanel(e: string | null): void;
}

interface ISearchFilters {
  [key: string]: {
    value: string;
    label: string;
  }[];
}
interface ISelectedFilters {
  [key: string]: any | object;
}

export interface IRangeFilter {
  value: string;
  label: string;
}

export interface ISlideoutModalProps extends IBaseProps {
  isOpen?: boolean;
  openPanel?: string | null;
  searchFilters: ISearchFilters;
  selectedFilters?: ISelectedFilters;
}

type TPanelData = {
  panelId: string;
  label: string;
  filterIds?: string[];
};

export interface IPanelsData {
  [key: string]: TPanelData;
}

interface IFilter {
  searchFilters: ISearchFilters;
  selectedFilters?: ISelectedFilters;
  setSelectedFiltersState(e: ISelectedFilters): void;
}

export interface IFilterMakeAndModel extends IFilter {}
export interface IFilterBudget extends IFilter {}
export interface IFilterCheckboxes extends IFilter {
  name: string;
}
