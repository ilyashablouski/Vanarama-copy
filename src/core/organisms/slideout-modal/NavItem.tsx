import React, { FC } from 'react';

import { INavItemProps } from './interfaces';

const NavItem: FC<INavItemProps> = props => {
  const { selected, label, onClick } = props;

  return (
    <button
      className="slideout-modal--nav-item"
      type="button"
      onClick={onClick}
    >
      <span className="slideout-modal--nav-item-label">{label}</span>
      {selected && (
        <span className="slideout-modal--nav-item-selected">
          {selected.join(', ')}
        </span>
      )}
    </button>
  );
};

export default React.memo(NavItem);
