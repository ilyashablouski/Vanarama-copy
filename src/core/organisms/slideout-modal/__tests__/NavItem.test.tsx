import React from 'react';
import { shallow } from 'enzyme';
import NavItem from '../NavItem';

import { INavItemProps } from '../interfaces';

const getComponent = (props: INavItemProps) => <NavItem {...props} />;

const onClick = jest.fn();

describe('<NavItem />', () => {
  it('should render correctly', () => {
    // ACT
    const wrapper = shallow(getComponent({ label: 'Label', onClick }));

    // ASSERT
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with selected', () => {
    const selected = ['One', 'Two', 'Three'];
    // ACT
    const wrapper = shallow(
      getComponent({
        label: 'Label',
        onClick,
        selected,
      }),
    );

    // ASSERT
    expect(wrapper).toMatchSnapshot();
  });
});
