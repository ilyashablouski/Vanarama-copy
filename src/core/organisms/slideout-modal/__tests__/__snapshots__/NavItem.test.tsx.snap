// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem /> should render correctly 1`] = `
<div>
  <button
    class="slideout-modal--nav-item"
    type="button"
  >
    <span
      class="slideout-modal--nav-item-label"
    >
      Label
    </span>
  </button>
</div>
`;

exports[`<NavItem /> should render correctly with selected 1`] = `
<div>
  <button
    class="slideout-modal--nav-item"
    type="button"
  >
    <span
      class="slideout-modal--nav-item-label"
    >
      Label
    </span>
    <span
      class="slideout-modal--nav-item-selected"
    >
      One, Two, Three
    </span>
  </button>
</div>
`;
